main.js

```diff
@@ -49,31 +49,43 @@ function validateUserInput() {
 
 function displayCards() {
   ideaCardGrid.innerHTML = "";
+  var shouldShowAllIdeas = showStarredIdeasButton.innerText === `Show Starred Ideas`;
+  var shouldShowStarredIdeas = showStarredIdeasButton.innerText === `Show All Ideas`;
   for (var i = 0; i < ideas.length; i++) {
-    var starSource;
-    if (ideas[i].star) {
-      starSource = "./assets/star-active.svg";
-    } else {
-      starSource = "./assets/star.svg";
+    if (shouldShowAllIdeas) {
+      ideaCardGrid.innerHTML += generateInnerHTML(ideas[i]);
+    } else if (shouldShowStarredIdeas && ideas[i].star) {
+      ideaCardGrid.innerHTML += generateInnerHTML(ideas[i]);
     }
-    ideaCardGrid.innerHTML +=
-    `<div class="box-container" id="${ideas[i].id}">
-      <div class="box-header-container">
-        <input class="star-btn" type="image" name="star button" src="${starSource}" alt="picture-of-a-star">
-        <input class="delete-btn" type="image" name="delete button" src="./assets/delete.svg" alt="picture-of-an-x">
-      </div>
-      <div class="title-body-container">
-        <label class="idea-title">${ideas[i].title}</label>
-        <div class="idea-body-container">
-          <p class="idea-body">${ideas[i].body}</p>
-        </div>
-      </div>
-      <div class="box-footer-container">
-        <button class="comment-btn"><img src="./assets/comment.svg"></button>
-        <label class="comment-label">Comment</label>
-      </div>
-    </div>`;
   }
+  if (searchBox.value) {
+    activeSearchFilter();
+  }
+}
+
+function generateInnerHTML(idea) {
+  var starSource;
+  if (idea.star) {
+    starSource = "./assets/star-active.svg";
+  } else {
+    starSource = "./assets/star.svg";
+  }
+  return `<div class="box-container" id="${idea.id}">
+    <div class="box-header-container">
+      <input class="star-btn" type="image" name="star button" src="${starSource}" alt="picture-of-a-star">
+      <input class="delete-btn" type="image" name="delete button" src="./assets/delete.svg" alt="picture-of-an-x">
+    </div>
+    <div class="title-body-container">
+      <label class="idea-title">${idea.title}</label>
+      <div class="idea-body-container">
+        <p class="idea-body">${idea.body}</p>
+      </div>
+    </div>
+    <div class="box-footer-container">
+      <button class="comment-btn"><img src="./assets/comment.svg"></button>
+      <label class="comment-label">Comment</label>
+    </div>
+  </div>`;
```

Comment:
I created this generateInnerHTML function for better readability inside of our displayCards function.